File: Data Structures & Algorithms/binary-search/submission-1.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        left = 0
        right = len(nums) - 1
        

        while left <= right:
            mid = left + (right - left)//2
            if nums[mid] == target :
                return mid
            elif nums[mid] < target :
                # if target is less than nums[mid] than we move greater portion
                left = mid + 1
            else :
                # if target is less than nums[mid] then move to left portion so 
                # right becomes behind mid 
                right = mid -1
        return -1

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-1.cpp
class Solution {
public:
    int findMin(vector<int> &nums) {
        int min_num = nums[0];
       for (int i = 0 ; i < nums.size(); i++){
        min_num = min(min_num,nums[i]);
       }
       return min_num;
    }
};
